Much better rewrite, generate install_manifest.text with scons

ryanthecoder opened the pull request:
Take advantage of an ENV variable we already had to create the manifest file

BST-263
http://jira.makerbot.net/browse/BST-263

Commit:
Much better rewrite, generate install_manifest.text with scons
Take advantage of an ENV variable we already had to create the manifest file

BST-263
http://jira.makerbot.net/browse/BST-263

tedbrandston commented on mb_install.py:
```diff
@@ -144,6 +144,9 @@ def mb_install_system(env, source, dest):
     return target
 
 def mb_create_install_target(env):
+    with open('install_manifest.txt', 'w') as fp:
```
`env.File('#/install_manifest.txt').abspath`

dudecc replied:
Aww, I liked that it went in obj/

tedbrandston replied:
hmm, I suppose you're right. Ryan was saying there was some problem with it being in obj when trying to access it. Also, obj only exists on some platforms.

@ryanthecoder what was wrong with obj?

ryanthecoder replied:
The Python based repositories don't have obj directories on any platform

Commit:
move the install manifest location to the root of the repo
not all repos use the obj directory, so for the sake of consitency
move the location of the generated install_manifest.txt out of the variant dir

BST-263
http://jira.makerbot.net/browse/BST-263

Commit:
fix an issue in recursive install where directories arn't detected

os.path.isdir() does not understand paths the start with '#'
soulution is to create an absoulte path using scons 'Entry' module

BST-263
http://jira.makerbot.net/browse/BST-263

Commit:
remove a debugging line from the code that wasn't supposed to be there

BST-263
http://jira.makerbot.net/browse/BST-263

Commit:
Fix where relative path was incorrect due to scons style paths
just had to switch it to the str_src constructed earlier

BST-263
http://jira.makerbot.net/browse/BST-263
